src/tests/ftest/util/dmg_utils.py

```diff
@@ -23,44 +23,183 @@
-        self.nvme = FormattedParameter("--nvme={}")
-        self.svcn = FormattedParameter("--svcn={}")
-        self.target = FormattedParameter("--target={}")
-        self.force = FormattedParameter("--force", False)
-        self.pool = FormattedParameter("--pool={}")
-        self.svc = FormattedParameter("--svc={}")
-        self.rank = FormattedParameter("--rank={}")
-        self.cont = FormattedParameter("--cont={}")
-        self.oid = FormattedParameter("--oid={}")
-
-    def get_param_names(self):
-        """Get a sorted list of dmg command parameter names."""
-        names = self.get_attribute_names(FormattedParameter)
-        names.extend(["request", "action"])
-        return names
+        self.insecure = FormattedParameter("-i", True)
+        self.debug = FormattedParameter("-d", False)
+        self.json = FormattedParameter("-j", False)
+
+    def _get_action_command(self):
+        """Assign a command object for the specified request and action."""
+        if self.action.value == "format":
+            self.action_command = self.DmgFormatSubCommand()
+        elif self.action.value == "prepare":
+            self.action_command = self.DmgPrepareSubCommand()
+
+    class DmgFormatSubCommand(CommandWithParameters):
+        """Defines an object representing a format sub dmg command."""
+
+        def __init__(self):
+            """Create a dmg Command object."""
+            super(DmgCommand.DmgFormatSubCommand, self).__init__(
+                "/run/dmg/format/*", "format")
+            self.force = FormattedParameter("-f", False)
+
+    class DmgPrepareSubCommand(CommandWithParameters):
+        """Defines a object representing a prepare sub dmg command."""
+
+        def __init__(self):
+            """Create a dmg Command object."""
+            super(DmgCommand.DmgPrepareSubCommand, self).__init__(
+                "/run/dmg/prepare/*", "prepare")
+            self.pci_wl = FormattedParameter("-w {}")
+            self.hugepages = FormattedParameter("-p {}")
+            self.targetuser = FormattedParameter("-u {}")
+            self.nvmeonly = FormattedParameter("-n", False)
+            self.scmonly = FormattedParameter("-s", False)
+            self.force = FormattedParameter("-f", False)
+            self.reset = FormattedParameter("--reset", False)
+
+
+def storage_scan(hosts):
+    """ Execute scan command through dmg tool to servers provided.
+
+    Args:
+        hosts (list): list of servers to run scan on.
+
+    Returns:
+        Avocado CmdResult object that contains exit status, stdout information.
+
+    """
+    # Create and setup the command
+    dmg = DmgCommand(get_file_path("bin/daos_shell"))
+    dmg.request.value = "storage"
+    dmg.action.value = "scan"
+    dmg.insecure.value = True
+    dmg.hostlist.value = hosts
+
+    try:
+        result = dmg.run()
+    except CommandFailure as details:
+        print("<dmg> command failed: {}".format(details))
+        return False
```

Comment:
To avoid returning different object types, this should either raise an exception at this point or return `None`.  This would indicate that no devices were found due to an error (None returned) vs. no devices where found with a successful command (empty list returned).

Reply:
Changed to return None.